pkg/driver/executor/executor.go

```diff
@@ -58,25 +58,39 @@ func NewExecutor(factory *client.Factory, config *api.MachineProviderConfig) (*E
 // CreateMachine creates a new OpenStack server instance and waits until it reports "ACTIVE".
 // If there is an error during the build process, or if the building phase timeouts, it will delete any artifacts created.
 func (ex *Executor) CreateMachine(ctx context.Context, machineName string, userData []byte) (string, error) {
-	serverNetworks, err := ex.resolveServerNetworks(machineName)
-	if err != nil {
-		return "", fmt.Errorf("failed to resolve server networks: %w", err)
+	var (
+		server *servers.Server
+		err    error
+	)
+
+	deleteOnFail := func(err error) error {
+		klog.Infof("attempting to delete server [Name=%q] after unsuccessful create operation with error: %v", machineName, err)
+		if errIn := ex.DeleteMachine(ctx, machineName, ""); errIn != nil {
+			return fmt.Errorf("error deleting server [Name=%q] after unsuccessful creation attempt: %v. Original error: %w", machineName, errIn, err)
+		}
+		return err
 	}
 
-	server, err := ex.deployServer(machineName, userData, serverNetworks)
-	if err != nil {
-		return "", fmt.Errorf("failed to deploy server for machine %q: %w", machineName, err)
+	server, err = ex.getMachineByName(ctx, machineName)
+	if err == nil {
+		klog.Infof("found existing server [Name=%q, ID=%q]", machineName, server.ID)
 	}
 
-	providerID := EncodeProviderID(ex.Config.Spec.Region, server.ID)
+	if err != nil && !errors.Is(err, ErrNotFound) {
```

Comment:
Is the check `err != nil` required here? I mean after we exit above if `err == nil` the `err` will always be unequal nil.

Reply:
I think it is because `!errors.Is(err, ErrNotFound)` will return false if the error is nil. So if you simply remove the `err!=nil` check you would always return on line 80 if the server is found.

We could change the condition with `else`:
```
if err == nil {
}  else if !errors.Is(err, ErrNotFound) {
} else{}
```
Does that make it more readable ?